### ISSUE TITLE
Feature request: export list.

### ISSUE THREAD

**Issue by degski** [Microsoft/vcpkg#4067]
The vcpkg list command currently exports something like this:

    abseil:x64-windows-static                          2018-07-30       an open-source collection designed to augment th...
    abseil:x86-windows-static                          2018-07-30       an open-source collection designed to augment th...
    aurora:x64-windows-static                          2017-06-21-c7... Aurora is an open-source C++ library providing v...
    benchmark:x86-windows-static                       1.4.1            A library to support the benchmarking of functio...
    boost-accumulators:x64-windows-static              1.67.0           Boost accumulators module
    boost-algorithm:x64-windows-static                 1.67.0           Boost algorithm module
    boost-align:x64-windows-static                     1.67.0           Boost align module
    boost-any:x64-windows-static                       1.67.0           Boost any module
    boost-array:x64-windows-static                     1.67.0           Boost array module
    boost-asio:x64-windows-static                      1.67.0-1         Boost asio module
    boost-assert:x64-windows-static                    1.67.0           Boost assert module
    ...

This is obviously useful for it's intended purpose, but I would like to request something like the `dir /b` option, i.e. a list that looks like:
    abseil
    aurora
    benchmark
    boost-accumulators
    boost-algorithm
    boost-align
    boost-any
    boost-array
    boost-asio
    boost-assert
    ...

This way I can export all the names of the packages to be installed if I would like to (or have to) recreate my complete local repo.

It is probably better to not have newlines, but just have it all on one line, a copy-past will do the job then.

**Comment by ras0219-msft** [comment]
Thanks for the suggestion, I definitely see it being useful for that!

Some clarifying questions:
- Should the list output triplets? If not, it would need to list packages only for a single triplet at a time.
- Would it be better to output the current information in an XML/json blob that could be sliced-and-diced however desired?

**Comment by degski** [comment]
@ras0219-msft 

No, I don't think it should output triplets, on the other hand, filtering them with triplets could be useful in my opinion, i.e. only get the x64-windows-static libs, otherwise just output them all, mixed. Ideally there would be one output per lib, basically the inverse of the `vcpkg install foo bar foobar` command. The idea would be that you now have a list, that you could save somewhere or copy immediately to install that same list like with another triplet.
xml/json is not my strong point, so it would not help me much, but maybe others.

**Comment by OlafvdSpek** [comment]
Some thoughts:
> This way I can export all the names of the packages to be installed if I would like to (or have to) recreate my complete local repo.

I put the manually maintained list in a vcpkg-install.txt file that I check into version control of a project, that way you can easily install all needed packages.

Exporting the current list might include more then the current project really needs.

**Comment by davidhunter22** [comment]
Actually I think the bulk install is the most useful part. Given a list of installed triplets in a file like the vcpkg-install.txt mentioned by  OlafvdSpek it would be nice to do
vcpkg install vcpkg-install.txt 
The format of this file could just be the existing output of "vcpkg list" and ignore the comment at the end. For bonus points it could then check that the installed version of each packaged matched that in the  output of vcpkg list.

**Comment by PhoebeHui** [comment]
@degski, May be you can try the following command:
```
 ./vcpkg list | % { @($_.split(":"))[0] }
```
